gosymbol_test.go

```diff
@@ -1 +1,49 @@
-package gosymbol_test
+package gosymbol
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestAddArgument(t *testing.T) {
+	type inputArgs struct {
+		args  Arguments
+		v     variable
+		value float64
+	}
+
+	tests := []struct {
+		input         inputArgs
+		expectedError error
+		expectedArgs  Arguments
+	}{
+		{ // Test 1: testing with duplicate name
+			input: inputArgs{
+				args:  Arguments{Var("X"): 1},
+				v:     Var("X"),
+				value: 2,
+			},
+			expectedError: errors.New("multiple variables have the same name"),
```

Comment:
Really nice with expected error! Should probably rewrite everything else to follow this pattern